packages/big-design/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -280,7 +286,13 @@ export const MultiSelect = typedMemo(
       inputId: id,
       inputValue,
       itemToString: (option) => (option ? option.content : ''),
-      items: filteredOptions,
+      items: [
+        ...(selectAll
+          ? // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+            ([{ content: localization?.selectAll, value: 'select-all' }] as Array<SelectOption<T>>)
+          : []),
+        ...filteredOptions,
+      ],
```

Comment:
🤔 I see why you need the `as` here, but I wonder if we should actually keep `value` as a generic and rather just type it as `string`.

I'm not sure it makes sense for value to be anything other than `string`, someone with more expertise in BD might correct me.

cc: @jorgemoya @chanceaclark

Reply:
I agree we should change this to string, I believe the original idea of having value as a generic is flawed, but maybe something we fix in a different PR.